R/modifiers.r

```diff
@@ -109,6 +109,9 @@ regex <- function(pattern, ignore_case = FALSE, multiline = FALSE,
 boundary <- function(type = c("character", "line_break", "sentence", "word"),
                     skip_word_none = TRUE, ...) {
   type <- match.arg(type)
+
+  if (type != "word" & missingArg(skip_word_none)) skip_word_none <- FALSE
```

Comment:
Maybe it would be better to make the default value of `skip_word_none` `NA`, and then do:

```R
if (identical(skip_word_none, NA)) {
  skip_word_none <- type == "word"
}
```

This would also need some doc updates